apps/cli/src/args.ts

```diff
@@ -45,4 +45,29 @@ export const CLI_ARGS_OPTIONS = {
     type: "boolean" as const,
     short: "h",
   },
+  type: {
+    type: "string" as const,
+    short: "t",
+    default: "remix-app-server",
+  },
+};
+
+export type File = {
+  name: string;
+  content: string;
+  encoding: "utf-8";
+};
+
+export type Folder = {
+  name: string;
+  files: File[];
+  subFolders: Folder[];
+};
+
+export type ProjectType = "defaults" | "vercel" | "remix-app-server";
```

Comment:
ProjectType is alright, but Destination or Platform or Framework would be more clear, kinda tryint to describe this thing more accurately. "defaults" is not very clear what it is but if that's the only option then maybe "default" is more correct. I would still prefer a name that is descriptive